### PR TITLE
[xcode9] More SceneKit bindings

### PR DESCRIPTION
**Pull request by migueldeicaza**

### CODE REVIEW

File: src/scenekit.cs [hunk 1]
```diff
@@ -98,6 +98,9 @@ interface MTLRenderPassDescriptor {}
 	[Mac (10,8), iOS (8,0)]
 	delegate void SCNSceneSourceStatusHandler (float /* float, not CGFloat */ totalProgress, SCNSceneSourceStatus status, NSError error, ref bool stopLoading);
 
+	delegate void SCNAnimationDidStartBlock (SCNAnimation animation, SCNAnimatable receiver);
```
Comment (dalexsoto) [inline src/scenekit.cs]: We should drop the `Block` suffix here and maybe append `Handler` instead.

File: src/scenekit.cs [hunk 2]
```diff
@@ -98,6 +98,9 @@ interface MTLRenderPassDescriptor {}
...
+	delegate void SCNAnimationDidStopBlock (SCNAnimation animation, SCNAnimatable receiver, bool completed);
```
Comment (dalexsoto) [inline src/scenekit.cs]: same

File: src/scenekit.cs [hunk 3]
```diff
@@ -152,7 +155,14 @@ interface SCNAnimatable {
 #if XAMCORE_2_0
 		[Abstract]
 #endif
-		[Mac (10,9)]
+		[Introduced (PlatformName.WatchOS, 3, 0)]
+		[Deprecated (PlatformName.WatchOS, 4, 0, message: "Use SCNAnimationPlayer.Paused] instead")]
```
Comment (dalexsoto) [inline src/scenekit.cs]: The message seems to have an unwanted typo `]` also I think our tests will complain about ending dot. You might want to use

> Use 'SCNAnimationPlayer.Paused' instead.

File: src/scenekit.cs [hunk 4]
```diff
@@ -152,7 +155,14 @@ interface SCNAnimatable {
...
+		[Introduced (PlatformName.TvOS, 9, 0)]
+		[Deprecated (PlatformName.TvOS, 11, 0,   message: "Use SCNAnimationPlayer.Paused] instead")]
```
Comment (dalexsoto) [inline src/scenekit.cs]: same and the ones below.

File: src/scenekit.cs [hunk 5]
```diff
@@ -4306,4 +4348,121 @@ interface SCNBufferStream {
...
+		bool Paused { get; set; }
+	
+		[Export ("play")]
+		void Play ();
+	
+		[Export ("stop")]
+		void Stop ();
+	
+		[Export ("stopWithBlendOutDuration:")]
+		void StopWithBlendOutDuration (double duration);
```
Comment (dalexsoto) [inline src/scenekit.cs]: I think we can just use just `Stop` here and use `blendOutDuration` as parameter name instead

File: src/scenekit.cs [hunk 6]
```diff
@@ -4306,4 +4348,121 @@ interface SCNBufferStream {
...
+		bool Paused { get; set; }
+	
+		[Export ("play")]
+		void Play ();
+	
+		[Export ("stop")]
+		void Stop ();
+	
+		[Export ("stopWithBlendOutDuration:")]
+		void StopWithBlendOutDuration (double seconds);
```
Comment (rolfbjarne) [inline src/scenekit.cs]: I think `void Stop (double blendOutDuration)` looks better.

Comment (spouliot) [reply src/scenekit.cs]: It looks better today but we can end up with new variations, like `stopWithBlendInDuration:`, in the future, the argument (`double`) is too easy to reuse for other purposes

**Pull request merged**